### PR TITLE
*: support using multiply bindings in planner

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

Support using multiply bindings in planner.

### What is changed and how it works?

First, we will find the best plan, after that it will loop over all valid bindings and generate plans, if the best plan is in it, we will directly use it; otherwise we will use the best cost plan among the bindings.

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - None
Release note

 - None

### CODE REVIEW

**Comment by lamxTyler** [comment]
@winoros Yes, if there are bindings, we will use the one with lowest cost; if not, we will use the best cost plan.

File: planner/optimize.go [hunk 1]
```diff
@@ -46,28 +45,58 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
...
+			binding.Status = bindinfo.Invalid
+			handleInvalidBindRecord(sctx, scope, bindinfo.BindRecord{
+				OriginalSQL: bindRecord.OriginalSQL,
+				Db:          bindRecord.Db,
+				Bindings:    []bindinfo.Binding{binding},
+			})
+			continue
+		}
+		// The best bestPlan is in baselines, just use it.
+		if bestPlanHint == plannercore.GenHintsFromPhysicalPlan(plan) {
```
Comment (eurekaka) [inline planner/optimize.go]: Can we just compare `bestPlanHint` with hints in `bindRecord`, and avoid the `optimize` for these bindings if it is found in the list?

Comment (lamxTyler) [reply planner/optimize.go]: Good point.

**Pull request merged**